src/map/use-map-content.ts

```diff
@@ -0,0 +1,195 @@
+import { MapboxGeoJSONFeature } from 'mapbox-gl';
+import { useMemo } from 'react';
+
+import { LayerName, layers } from '../config/layers';
+import { BackgroundName } from '../config/types';
+import { ViewName, views } from '../config/views';
+
+/**
+ * get map style and layers definition based on:
+ * - selected background
+ * - selected layers, filters
+ * - selected data visualisation
+ * - any highlights / selections
+ */
+
+export interface MapParams {
+  background: BackgroundName;
+  view: ViewName;
+  dataLayerSelection: Record<LayerName, boolean>;
+  highlightedFeature: MapboxGeoJSONFeature;
+}
+
+function getMapSources(highlightedFeature: MapboxGeoJSONFeature) {
+  const res = {
+    satellite: {
+      type: 'raster',
+      url: 'mapbox://mapbox.satellite',
+      tileSize: 256,
+    },
+    light: {
+      type: 'raster',
+      tiles: [
+        'https://tiles.basemaps.cartocdn.com/light_nolabels/{z}/{x}/{y}.png',
+        'https://a.basemaps.cartocdn.com/light_nolabels/{z}/{x}/{y}.png',
+        'https://b.basemaps.cartocdn.com/light_nolabels/{z}/{x}/{y}.png',
+        'https://c.basemaps.cartocdn.com/light_nolabels/{z}/{x}/{y}.png',
+        'https://d.basemaps.cartocdn.com/light_nolabels/{z}/{x}/{y}.png',
+      ],
+      tileSize: 256,
+    },
+
+    road_edges: {
+      type: 'vector',
+      url: 'http://localhost:8080/data/road_edges.json',
```

Comment:
Will this need to change when not running locally? If so, can we skip the hostname here and below and just write `url: '/data/road_edges.json`? Or use some config (e.g. environment variable) to set it for local/staging/production?

Reply:
Agreed, need to introduce a configuration variable and build the URLs dynamically. A lot of this particular code for managing sources/layers can still change so I'd suggest to keep as-is for now during development and add the URL configuration once it crystallises more.